### ISSUE TITLE
pubsub/kafkapubsub: consumer group not rebalancing

### ISSUE THREAD

**Issue by burdiyan** [google/go-cloud#2435]
### Describe the bug

Having multiple consumers within the same consumer group, rebalance is not handled properly.

### To Reproduce

Having Kafka running: 

1. Create a topic with 2 or more partitions (below `backblaze_smart`).
2. Create a `main` package with the following code:

```go
package main

import (
	"context"
	"fmt"
	"os"
	"os/signal"
	"syscall"

	"gocloud.dev/pubsub/kafkapubsub"
)

var (
	brokers = []string{"localhost:9092"}
	topics  = []string{"backblaze_smart"}
)

func main() {
	if err := run(); err != nil {
		panic(err)
	}
}

func run() error {
	ctx, cancel := context.WithCancel(context.Background())
	defer cancel()

	go func() {
		ch := make(chan os.Signal, 1)
		signal.Notify(ch, os.Interrupt, os.Kill, syscall.SIGTERM)
		<-ch
		cancel()
		signal.Stop(ch)
	}()

	sub, err := kafkapubsub.OpenSubscription(
		brokers,
		kafkapubsub.MinimalConfig(),
		"pubsub-group-test",
		topics,
		&kafkapubsub.SubscriptionOptions{
			KeyName: "key",
		},
	)
	if err != nil {
		return err
	}
	defer sub.Shutdown(context.Background())

	for {
		msg, err := sub.Receive(ctx)
		if err != nil {
			return err
		}

		fmt.Printf("%s:%s\n", msg.Metadata["key"], string(msg.Body))
		msg.Ack()
	}
}
```

3. Start one instance of the program and see messages being printed in the console.
4. Start another instance of the program, and see messages being printed in the console of the second instance, but first instance stops.

## Expected behavior

Messages being printed in both consoles, since this topic has 2 partitions, and messages should spread across 2 instances.

### Version

https://github.com/google/go-cloud/commits/2e278191e064c358d4fee5cc10de4076878ad974

**Comment by vangent** [comment]
Hi, thanks for the report. Confirmed that this is a bug; fix is on the way.

**Comment by burdiyan** [comment]
@vangent Could you leave the issue open until the PR is merged? I guess otherwise it's a little bit confusing. If that goes against your contributing guidelines, sorry for that and nevermind :)

**Comment by vangent** [comment]
HI @burdiyan -- sorry for the confusion. The bug should be fixed via #2445 -- it'd be great if you could verify. I have another PR (#2451) out to add a better test, which I'm also associating with this issue, but is just a test so shouldn't block you.
